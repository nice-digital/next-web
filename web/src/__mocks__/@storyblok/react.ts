export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
