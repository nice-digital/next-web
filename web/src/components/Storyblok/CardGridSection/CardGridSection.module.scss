@use '~/styles/colours';
@use '~/styles/spacing' as *;
@use '@nice-digital/nds-core/scss/layout';
@use '@nice-digital/nds-core/scss/spacing';
@use '@nice-digital/nds-core/scss/typography';
@use '@nice-digital/nds-core/scss/utils';

.cardGridSection {
  @include layout.fullWidth();
  background-color: colours.$card-section-background;
  padding: $section-medium-spacing 0;

  &__container {
    @include layout.fullWidthContainer();
  }

  &__intro {
    margin-bottom: utils.rem(spacing.$large);
  }

  &__heading {
    margin-top: 0;
  }

  &__leadText {
    p {
      @include typography.font-size(lead);
    }
  }

  &__leadTextAdditional {
    p {
      @include typography.font-size(p);
    }
  }

  &__gridRow {
    row-gap: utils.rem(spacing.$large);

    &:not(:last-child) {
      margin-bottom: utils.rem(spacing.$x-large);
    }
  }

  &__gridItem {
    > * {
      margin-bottom: 0; // Some components have their own margin-bottom which messes with grid layout
    }
  }

  &--largeSpacing {
    padding: $section-large-spacing 0;
  }

  &--smallSpacing {
    padding: $section-small-spacing 0;
  }

  &--transparent {
    background: transparent;
  }
}
