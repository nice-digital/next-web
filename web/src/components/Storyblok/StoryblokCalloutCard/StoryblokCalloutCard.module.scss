@use '@nice-digital/nds-core/scss/media-queries';
@use '@nice-digital/nds-core/scss/spacing';
@use '@nice-digital/nds-core/scss/utils';

.calloutImageContainer {
  aspect-ratio: 16 / 9;
  background-position: center;
  background-size: cover;
  display: block;
  flex: 0 0 auto;
  margin-bottom: utils.rem(spacing.$medium);
  min-width: 100%;
  object-fit: cover;
  
  @include media-queries.mq($from: sm) {
    aspect-ratio: 21 / 9;
  }
  
  @include media-queries.mq($from: md) {
    aspect-ratio: 16 / 9;
  }
}
