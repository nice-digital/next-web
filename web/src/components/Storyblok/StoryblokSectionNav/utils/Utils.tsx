import { SBLink } from "@/types/SBLink";
import { fetchLinks } from "@/utils/storyblok";

export type ExtendedSBLink = SBLink & {
	childLinks?: ExtendedSBLink[];
};

export const getCurrentFolderItems = async (
	parentID: number
): Promise<ExtendedSBLink[]> => {
	// Get all items in current folder (current page and its siblings or, if current page is a root page, current page and its children)
	return await fetchLinks({
		with_parent: parentID,
	});
};

export const assignChildrenToParent = (
	currentFolderItems: ExtendedSBLink[],
	parents: ExtendedSBLink[]
): ExtendedSBLink[] => {
	for (const parent of parents) {
		const children = currentFolderItems.filter(
			(childLink) =>
				childLink.parent_id === parent.id && !childLink.is_startpage
		);

		parent.childLinks = children.length > 0 ? children : [];
	}
	return parents;
};

export const siblingHasChildrenFunction = async (
	slug: string,
): Promise<boolean> => {
	//const currentFolderItems = await getCurrentFolderItems(parentID);
	const startsWithCurrentFolderSlugItems = await fetchLinks({
		starts_with: slug,
	});

	// If any items in current folder have children, except for item corresponding with current page, return true
	const siblingHasChildren = startsWithCurrentFolderSlugItems.some(
		(item) => item.is_folder && item.slug !== slug
	);

	// console.log({ slug, siblingHasChildren });
	return siblingHasChildren;
};

export const fetchAndBuildParentAndChildTree = async (
	slug: string,
	parentID: number,
	children?: ExtendedSBLink[]
): Promise<ExtendedSBLink[]> => {
	//console.trace("fetchAndBuildParentAndChildTree called");
	//console.log("fetchAndBuildParentAndChildTree called with:", { slug, parentID });
	const currentFolderItems = await getCurrentFolderItems(parentID);

	// Get all items starting with the current folder slug - this is the only way to access the folder object
	const startsWithCurrentFolderSlugItems = await fetchLinks({
		starts_with: slug,
	});

	// Get currentFolder object so we can access its parent_id
	const currentFolder: ExtendedSBLink | undefined =
		startsWithCurrentFolderSlugItems.find(
			(item: ExtendedSBLink) => item.is_folder && item.slug === slug
		);

	// Check if any items in current folder, except for item corresponding with current page, have children
	// this code stops iterating as soon as it finds a matching element and is not asynchronous
	// //console.log({slug, startsWithCurrentFolderSlugItems})
	// const siblingHasChildrenBool = startsWithCurrentFolderSlugItems.some(
	// 	(item) => item.is_folder && item.slug !== slug
	// );

	const siblingHasChildrenBool = await siblingHasChildrenFunction(slug);
	// //console.log({ slug, siblingHasChildren });

	let tree: ExtendedSBLink[] = [];

	// If no current folder found or current folder has no parent_id, return the currentFolderItems
	if (!currentFolder || !currentFolder.parent_id) return currentFolderItems;

	// Get all items in current folder's parent folder (current "page" and its siblings OR parent page and its siblings, depending on current position in tree)
	const parentFolderItems = await fetchLinks({
		with_parent: currentFolder.parent_id,
	});
	tree = parentFolderItems;

	// If current page has no children, and none of current page's siblings have children, use current level and level above to create the parent/child tree structure

	if (!(children && children.length > 0 && siblingHasChildrenBool)) {
		console.log(
			`No current page children, no current page siblings children, building tree, children: ${children}, children.length: ${children?.length}, siblingHasChildren:${siblingHasChildrenBool}` )
		//console.trace("Conditional block executed: No current page children, no current page siblings children");
		//console.log("No current page children, no current page siblings children, building tree");
		tree = assignChildrenToParent(currentFolderItems, parentFolderItems);
	}

	return tree;
};

export const fetchCurrentAndParentFolderItems = async (
	parentID: number,
	slug: string
): Promise<{
	currentAndParentFolderItems: ExtendedSBLink[];
}> => {
	const currentAndParentFolderItemsOrCurrentFolderItems =
		await fetchAndBuildParentAndChildTree(slug, parentID);
	return {
		currentAndParentFolderItems:
			currentAndParentFolderItemsOrCurrentFolderItems,
	};
};

export const buildTree = async (
	parentID: number,
	slug: string,
	isRootPage: boolean | undefined
): Promise<ExtendedSBLink[]> => {
	//console.trace("buildTree called");
	//console.log("buildTree called with:", { parentID, slug });
	let tree: ExtendedSBLink[] = [];
	const currentFolderItems = await getCurrentFolderItems(parentID);
	const { currentAndParentFolderItems } =
		await fetchCurrentAndParentFolderItems(parentID, slug);
	tree = assignChildrenToParent(
		currentFolderItems,
		currentAndParentFolderItems
	);

	const currentPage = tree.find((item) => item.slug === slug);

	const currentPageOrSiblingHasNoChildren =
		!currentPage?.childLinks || currentPage?.childLinks.length === 0 || !siblingHasChildrenFunction;

	if (currentPage && currentPageOrSiblingHasNoChildren) {
		const currentFolderSlug = isRootPage
			? slug
			: slug.split("/").slice(0, -1).join("/");

		tree = await fetchAndBuildParentAndChildTree(
			currentFolderSlug,
			parentID,
			currentPage.childLinks ?? []
		);
	}

	return tree;
};
