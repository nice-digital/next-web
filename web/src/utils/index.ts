import slugify from "@sindresorhus/slugify";
import dayjs from "dayjs";

import { Project, ProjectStatus } from "@/feeds/inDev/types";
import {
	ProductGroup,
	ProductLite,
	ProductTypeAcronym,
} from "@/feeds/publications/types";

import { dateFormat, dateFormatShort } from "./constants";

/**
 * Returns a a sortable date with just year, month and date (without time).
 * We never expose times to users so it makes sense to just sort by date instead.
 *
 * @param isoDateStr The full date string with timezone e.g. `2021-05-21T10:54:52.4655487`
 * @returns The sortable date, without the timezone e.g. `2021-05-21`
 */
export const stripTime = (isoDateStr: string): string =>
	isoDateStr.substring(0, isoDateStr.indexOf("T"));

/**
 * Formats the given ISO date string into a standard NICE string format
 *
 * @param isoDateStr the ISO date string for format
 * @param short whether to format the date in the standard short format
 * @returns The formatted date
 */
export const formatDateStr = (isoDateStr: string, short = false): string =>
	dayjs(isoDateStr).format(short ? dateFormatShort : dateFormat);

/**
 * Gets the path, relative to the root, of an indevelopment project overview page.
 *
 * Note: advice projects don't have project pages, so return null.
 *
 * @param project The indev project for which to get the URL
 * @returns The path of the project, relative to the root
 */
export const getProjectPath = (project: Project): string | null =>
	project.projectGroup == ProductGroup.Advice
		? null
		: project.status === ProjectStatus.Proposed
		? `/guidance/awaiting-development/${project.reference.toLowerCase()}`
		: `/guidance/indevelopment/${project.reference.toLowerCase()}`;

/**
 * Gets the path, relative to the root, of an published product overview page.
 *
 * @param product The product for which to get the URL
 * @returns The path of the product, relative to the root
 */
export const getProductPath = (
	product: Pick<ProductLite, "productGroup" | "id" | "productType" | "title">
): string => {
	let rootPath: string,
		productSlug = product.id.toLowerCase();

	switch (product.productGroup) {
		case ProductGroup.Guideline:
		case ProductGroup.Guidance:
		case ProductGroup.Standard:
			rootPath = "guidance";
			break;
		case ProductGroup.Advice:
			rootPath = "advice";
			break;
		case ProductGroup.Corporate:
			// There are 2 types of corporate products that have different URLs: corporate vs process and methods
			rootPath =
				product.productType === ProductTypeAcronym.ECD
					? "corporate"
					: "process";
			break;
		case ProductGroup.Other:
			if (product.productType !== ProductTypeAcronym.IND)
				throw Error(
					`Unsupported 'other' product type of ${product.productType}`
				);
			rootPath = "indicators";
			productSlug += `-${slugify(product.title)}`;
			break;
		default:
			throw `Unsupported product group ${product.productGroup} ${JSON.stringify(
				product
			)}`;
	}

	return `/${rootPath}/${productSlug}`;
};
