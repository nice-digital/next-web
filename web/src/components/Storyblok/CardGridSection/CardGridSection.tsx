import classnames from "classnames";

import { CardGrid } from "@/components/Storyblok/CardGrid/CardGrid";
import { StoryblokRichText } from "@/components/Storyblok/StoryblokRichText/StoryblokRichText";
import {
	CardGridSectionStoryblok,
	CardGridRowTestimonialsStoryblok,
	CardGridRowBasicStoryblok,
	CardGridRowCalloutStoryblok,
	CardGridRowCalloutWithImageStoryblok,
} from "@/types/storyblok";
import { fieldHasValidContent } from "@/utils/storyblok";

import styles from "./CardGridSection.module.scss";

export interface CardGridSectionProps {
	blok: CardGridSectionStoryblok | CardGridRowTestimonialsStoryblok;
}

export const CardGridSection: React.FC<CardGridSectionProps> = ({
	blok,
}: CardGridSectionProps) => {
	const {
		heading,
		headingLevel,
		leadText,
		secondaryLeadText,
		verticalPadding = "medium",
		theme = "subtle",
		cards: rows,
	} = blok;

	const HeadingElement = `h${headingLevel}` as keyof JSX.IntrinsicElements;
	const themeClass = `cardGridSection--${theme}`;
	const verticalPaddingClass = `cardGridSection--${verticalPadding}Spacing`;
	const leadTextProcessed =
		leadText && fieldHasValidContent(leadText) ? leadText : null;
	const secondaryLeadTextProcessed =
		secondaryLeadText && fieldHasValidContent(secondaryLeadText)
			? secondaryLeadText
			: null;

	return (
		<section
			className={classnames(
				styles.cardGridSection,
				styles[verticalPaddingClass],
				styles[themeClass]
			)}
			data-testid="card-grid-section"
		>
			<div className={styles.cardGridSection__container}>
				{heading || leadTextProcessed ? (
					<div className={styles.cardGridSection__intro}>
						{heading && (
							<HeadingElement className={styles.cardGridSection__heading}>
								{heading}
							</HeadingElement>
						)}
						{leadTextProcessed && (
							<StoryblokRichText
								content={leadTextProcessed}
								className={styles.cardGridSection__leadText}
							/>
						)}
						{leadTextProcessed && secondaryLeadTextProcessed && (
							<StoryblokRichText
								content={secondaryLeadTextProcessed}
								className={styles.cardGridSection__leadTextSecondary}
							/>
						)}
					</div>
				) : undefined}
				{rows.map(
					(
						row:
							| CardGridRowTestimonialsStoryblok
							| CardGridRowBasicStoryblok
							| CardGridRowCalloutStoryblok
							| CardGridRowCalloutWithImageStoryblok
					) => (
						<CardGrid row={row} key={row._uid} />
					)
				)}
			</div>
		</section>
	);
};
