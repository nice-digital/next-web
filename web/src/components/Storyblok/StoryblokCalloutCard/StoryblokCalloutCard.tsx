import React from "react";
import { Card, type CardHeadingLinkProps } from "@nice-digital/nds-card";
import Link from "next/link";
import { StoryblokImage } from "../StoryblokImage/StoryblokImage";
import { CalloutCardStoryblok, CalloutCardWithImageStoryblok } from "@/types/storyblok";
import { resolveStoryblokLink } from "@/utils/storyblok";

export interface StoryblokCalloutCardProps {
	blok: CalloutCardStoryblok | CalloutCardWithImageStoryblok;
}

export const StoryblokCalloutCard: React.FC<StoryblokCalloutCardProps > = ({
	blok,
}: StoryblokCalloutCardProps) => {
	const { heading, body, link, image } = blok;
	console.log("blok", blok);
	let cardLink: CardHeadingLinkProps | undefined = undefined;
	const resolvedLink = link ? resolveStoryblokLink(link) : undefined;
	console.log("resolved link", resolvedLink);
	if (resolvedLink?.url) {
		cardLink = {
			destination: resolvedLink.url,
			elementType: resolvedLink.isInternal ? Link : "a",
			method: "href",
		};
	}
	console.log("link",link);
	
	const renderImage = () => {
		return image ? (
			<StoryblokImage
				src={image?.filename ? image.filename : undefined}
				alt={image?.alt || "callout card image"}
				serviceOptions={{ height: 0, quality: 80, width: 708 }}
			/>
		) : null;
	};

	return (
		<div>
			<Card
				headingText={heading}
				callout
				link={cardLink || undefined}
				image={renderImage()}
			>
				{body}
			</Card>
		</div>
	);
};
