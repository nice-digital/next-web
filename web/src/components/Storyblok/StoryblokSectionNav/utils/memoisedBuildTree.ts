import mem from "mem";
import { buildTree, ExtendedSBLink } from "./Utils";
import { logger } from "@/logger";

export const memoisedBuildTree = mem(
  async (
    parentID: number,
    slug: string,
    isRootPage?: boolean
  ): Promise<ExtendedSBLink[]> => {
    logger.warn(`Calling ORIGINAL buildTree (not memoised) for slug: "${slug}", parentID: ${parentID}, isRootPage: ${isRootPage}`);
    return await buildTree(parentID, slug, isRootPage);
  },
  {
    maxAge: 1000 * 60 * 2, // 2 minutes
    cacheKey: ([parentID, slug, isRootPage]: [number, string, boolean?]) => {
      const key = `${parentID}_${slug}_${isRootPage ?? false}`;
      logger.warn(`Generated buildTree cache key: ${key}`);
      return key;
    },
  }
);
