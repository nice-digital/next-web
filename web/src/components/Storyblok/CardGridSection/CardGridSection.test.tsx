import { render, screen } from "@testing-library/react";

import {
	type TestimonialGridItemStoryblok,
	type CalloutCardStoryblok,
	type CalloutCardWithImageStoryblok,
	type CardStoryblok,
	type CardGridSectionStoryblok,
	type RichtextStoryblok,
} from "@/types/storyblok";
import { type StoryblokCalloutCardProps } from "../StoryblokCalloutCard/StoryblokCalloutCard";

import { resolveStoryblokLink } from "@/utils/storyblok";
import { CardGridSection, CardGridSectionProps } from "./CardGridSection";

jest.mock("@/utils/storyblok", () => ({
	resolveStoryblokLink: jest.fn(),
	fieldHasValidContent: jest.fn(() => true),
	constructStoryblokImageSrc: jest.fn((filename) => filename),
}));

const mockLeadText: RichtextStoryblok = {
	type: "doc",
	content: [
		{
			type: "paragraph",
			content: [
				{
					type: "text",
					text: "This is lead text.",
				},
			],
		},
	],
};

const cardLinkUrl = "https://nice.org.uk/guidance/ta10";

const mockCardProps = {
	blok: {
		component: "card",
		_uid: "1",
		heading: "Mock card title",
		body: "Mock card summary",
		link: {
			fieldtype: "multilink",
			id: "link1",
			url: cardLinkUrl,
			cached_url: cardLinkUrl,
			linktype: "url",
		},
	} as CardStoryblok,
};

const mockCalloutCardProps: StoryblokCalloutCardProps = {
	blok: {
		...mockCardProps.blok,
		component: "calloutCard",
	} as CalloutCardStoryblok,
};

const mockCalloutCardWithImageProps = {
	blok: {
		...mockCalloutCardProps.blok,
		image: {
			id: 18214711,
			alt: "",
			name: "",
			focus: "",
			title: "",
			source: "",
			filename: "test.JPG",
			copyright: "",
			fieldtype: "asset",
			meta_data: {},
			is_external_url: false,
		},
		component: "calloutCardWithImage",
	} as CalloutCardWithImageStoryblok,
};

describe("CardGridSection", () => {
	beforeEach(() => {
		(resolveStoryblokLink as jest.Mock).mockImplementation((link) => ({
			url: link?.url || link?.cached_url,
			isInternal:
				(link?.url?.startsWith("/") || link?.cached_url?.startsWith("/")) ??
				false,
		}));
	});

	afterEach(() => {
		jest.clearAllMocks();
	});

	const mockProps: CardGridSectionProps = {
		blok: {
			component: "cardGridSection",
			_uid: "section1",
			heading: "Card Grid Section Heading",
			headingLevel: "2",
			leadText: mockLeadText,
			theme: "transparent",
			verticalPadding: "small",
			cards: [
				{
					component: "cardGridRowBasic",
					columns: "3",
					gridItems: [
						{
							...mockCardProps.blok,
						},
						{
							...mockCardProps.blok,
						},
						{
							...mockCardProps.blok,
						},
						{
							...mockCardProps.blok,
						},
					] as CardStoryblok[],
					_uid: "row1",
				},
			],
		} as CardGridSectionStoryblok,
	};

	it("renders without crashing", () => {
		render(<CardGridSection {...mockProps} />);
		expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent(
			"Card Grid Section Heading"
		);
	});

	it("renders lead text correctly", () => {
		render(<CardGridSection {...mockProps} />);
		const leadTextElement = screen.getByTestId("storyblok-rich-text");
		expect(leadTextElement).toBeInTheDocument();
		expect(leadTextElement).toHaveTextContent("This is lead text.");
	});

	it("applies correct theme class", () => {
		render(<CardGridSection {...mockProps} />);
		const section = screen.getByTestId("card-grid-section");
		expect(section).toHaveClass("cardGridSection--transparent");
	});

	it("renders the correct grid structure", () => {
		render(<CardGridSection {...mockProps} />);
		const grid = screen
			.getByTestId("card-grid-section")
			// eslint-disable-next-line testing-library/no-node-access
			.querySelector("div.grid, ul.grid");
		expect(grid).toBeInTheDocument();
	});

	it("resolves links correctly", () => {
		render(<CardGridSection {...mockProps} />);
		const linkElement = screen.getAllByRole("link", {
			name: "Mock card title",
		});
		expect(linkElement[0]).toBeInTheDocument();
		expect(linkElement[0]).toHaveAttribute("href", cardLinkUrl);
	});

	it("renders Card content correctly", () => {
		render(<CardGridSection {...mockProps} />);
		const cardList = screen.getByRole("list");
		const cardListItems = screen.getAllByRole("listitem");
		const card = screen.getAllByText("Mock card title");
		expect(cardList).toBeInTheDocument();
		expect(cardListItems.length).toBe(4);
		expect(cardListItems[0]).toHaveTextContent("Mock card title");
		expect(card.length).toBe(4);
	});

	it("renders Callout Card content correctly", () => {
		const mockCalloutProps: CardGridSectionProps = {
			blok: {
				...mockProps.blok,
				cards: [
					{
						component: "cardGridRowCallout",
						columns: "3",
						gridItems: [
							{
								...mockCalloutCardProps.blok,
							},
							{
								...mockCalloutCardProps.blok,
							},
							{
								...mockCalloutCardProps.blok,
							},
						] as CalloutCardStoryblok[],
						_uid: "row1",
					},
				],
			},
		};
		render(<CardGridSection {...mockCalloutProps} />);
		const cardList = screen.getByRole("list");
		const cardListItems = screen.getAllByRole("listitem");
		const calloutCard = screen.getAllByText("Mock card title");
		expect(cardList).toBeInTheDocument();
		expect(cardListItems.length).toBe(3);
		expect(cardListItems[0]).toHaveTextContent("Mock card title");
		expect(calloutCard.length).toBe(3);
	});

	it("renders Callout Card With Image content correctly", () => {
		const mockCalloutProps: CardGridSectionProps = {
			blok: {
				...mockProps.blok,
				cards: [
					{
						component: "cardGridRowCalloutWithImage",
						columns: "2",
						gridItems: [
							{
								...mockCalloutCardWithImageProps.blok,
							},
							{
								...mockCalloutCardWithImageProps.blok,
							},
						] as CalloutCardWithImageStoryblok[],
						_uid: "row1",
					},
				],
			},
		};
		render(<CardGridSection {...mockCalloutProps} />);
		const cardList = screen.getByRole("list");
		const cardListItems = screen.getAllByRole("listitem");
		const calloutCard = screen.getAllByText("Mock card title");
		expect(cardList).toBeInTheDocument();
		expect(cardListItems.length).toBe(2);
		expect(cardListItems[0]).toHaveTextContent("Mock card title");
		expect(calloutCard.length).toBe(2);
	});

	it("renders Testimonial content correctly", () => {
		const mockTestimonialProps: CardGridSectionProps = {
			blok: {
				...mockProps.blok,
				cards: [
					{
						component: "cardGridRowTestimonials",
						columns: "1",
						gridItems: [
							{
								component: "testimonialGridItem",
								_uid: "1",
								filename: "test-file",
								quoteName: "Test Name",
								quoteRole: "Test Role",
								quoteText: "Test Quote Text",
								image: {
									id: 123,
									filename: "test-image-filename",
									alt: "test-alt",
									fieldtype: "asset",
									name: "test-name",
									title: "test-title",
									focus: null,
								},
								variant: "default",
							} as TestimonialGridItemStoryblok,
						],
						_uid: "row1",
					},
				],
			},
		};
		render(<CardGridSection {...mockTestimonialProps} />);
		const testimonial = screen.getByText("Test Quote Text");
		expect(testimonial).toBeInTheDocument();
		const cardList = screen.queryByRole("list");
		const cardListItems = screen.queryByRole("listitem");
		expect(cardList).not.toBeInTheDocument();
		expect(cardListItems).not.toBeInTheDocument();
	});

	it("matches snapshot", () => {
		const { container } = render(<CardGridSection {...mockProps} />);
		expect(container).toMatchSnapshot();
	});
});
