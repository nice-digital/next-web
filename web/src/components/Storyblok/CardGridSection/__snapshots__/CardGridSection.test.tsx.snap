// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGridSection matches snapshot 1`] = `
<div>
  <section
    class="cardGridSection cardGridSection--mediumSpacing cardGridSection--subtle"
    data-testid="card-grid-section"
  >
    <div
      class="cardGridSection__container"
    >
      <div
        class="cardGridSection__intro"
      >
        <h2
          class="cardGridSection__heading"
        >
          Card Grid Section Heading
        </h2>
        <div
          class="storyblokRichTextWrapper cardGridSection__leadText"
          data-testid="storyblok-rich-text"
        >
          <p>
            This is lead text.
          </p>
        </div>
        <div
          class="storyblokRichTextWrapper cardGridSection__leadTextSecondary"
          data-testid="storyblok-rich-text"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut felis leo, fermentum quis posuere non, viverra at nulla.
          </p>
        </div>
      </div>
      <ul
        class="grid grid--equal-height grid--loose cardGrid__cardGridRow"
        data-testid="card-grid"
      >
        <li
          class="cardGrid__cardGridItem"
          data-component="grid-item"
          data-g="12 md:4"
        >
          <article
            class="card card--flush"
            data-component="card"
          >
            <header
              class="card__header"
            >
              <p
                class="card__heading"
              >
                <a
                  href="https://nice.org.uk/guidance/ta10"
                >
                  Mock card title
                </a>
              </p>
            </header>
            <p
              class="card__summary"
            >
              Mock card summary
            </p>
          </article>
        </li>
        <li
          class="cardGrid__cardGridItem"
          data-component="grid-item"
          data-g="12 md:4"
        >
          <article
            class="card card--flush"
            data-component="card"
          >
            <header
              class="card__header"
            >
              <p
                class="card__heading"
              >
                <a
                  href="https://nice.org.uk/guidance/ta10"
                >
                  Mock card title
                </a>
              </p>
            </header>
            <p
              class="card__summary"
            >
              Mock card summary
            </p>
          </article>
        </li>
        <li
          class="cardGrid__cardGridItem"
          data-component="grid-item"
          data-g="12 md:4"
        >
          <article
            class="card card--flush"
            data-component="card"
          >
            <header
              class="card__header"
            >
              <p
                class="card__heading"
              >
                <a
                  href="https://nice.org.uk/guidance/ta10"
                >
                  Mock card title
                </a>
              </p>
            </header>
            <p
              class="card__summary"
            >
              Mock card summary
            </p>
          </article>
        </li>
        <li
          class="cardGrid__cardGridItem"
          data-component="grid-item"
          data-g="12 md:4"
        >
          <article
            class="card card--flush"
            data-component="card"
          >
            <header
              class="card__header"
            >
              <p
                class="card__heading"
              >
                <a
                  href="https://nice.org.uk/guidance/ta10"
                >
                  Mock card title
                </a>
              </p>
            </header>
            <p
              class="card__summary"
            >
              Mock card summary
            </p>
          </article>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
